proposed/2023/project-exclusion-for-transitive-dependencies.md

```diff
@@ -0,0 +1,130 @@
+```
+warn : The command option '--exclude-project' is ignored by this command.
+
+The following sources were used:
+   https://api.nuget.org/v3/index.json
+
+The given project `projectA` has no vulnerable packages given the current sources.
+The given project `projectB` has no vulnerable packages given the current sources.
+```
+
+### Technical explanation
+
+Currently, the implementation determines a [ReportType](https://github.com/NuGet/NuGet.Client/blob/d9d5a38c6e0c3ccc36b1b21cb8fa468474690080/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ReportType.cs#L6) based on the presence of options:
+| Option              | Resulting ReportType  |
+| ------------------- | --------------------- |
+| --deprecated        | ReportType.Deprecated |
+| --outdated          | ReportType.Outdated   |
+| --vulnerable        | ReportType.Vulnerable |
+| _none of the above_ | ReportType.Default    |
+
+MSBuild version resolver is then called with the `includeProject` parameter set to `false` for all values except `Default`, where it is set to `true`.
+
+The implementation would consist or changing the value to `false` when the `exclude-project` option is set.
+
+## Drawbacks
+
+It adds a new option to a command, which could result in it being more difficult to use.
+
+## Rationale and alternatives
+
+This design is retro compatible with the previous behavior of the command. It also limits the output of the command to the expected results.
+
+Other approaches could have been taken:
+* Adding a `--include-project` option and modifying the default behavior of the list command: this would have made the whole command consistent, whether a `--deprecated`, `--vulnerable`, or `--outdated` option is passed or not. However, this would create a breaking change with previous behavior
+* Adding the type of dependency to the output: this would have made the output filterable afterwards, without adding flags. However, this could break console output parsing - which could be acceptable -, and could force an `output-version` increment as the machine-readable format would change. It would also require an active filtering process after the command is run. Whether this proposal is implemented, this could be an interesting option if consequences are accepted.
+
+If we do not implement this option, then users of the command will have to be able to filter projects out of the output. However, this is not an easy task, and it is subject to errors if for example a project and a package have the same name.
+
+## Prior Art
+
+* [Machine readable JSON output for dotnet list package](https://github.com/NuGet/Home/blob/a6fea29b47359df9a1a03c381459e2b72c6062e9/proposed/2022/DotnetListPackageMachineReadableJsonOutput.md)
+
+## Unresolved Questions
+
+* Should the nature of the dependency (project or package) be included in the command output (or at least in the machine-readable one) ?
```

Comment:
IMO, I think that might be a miss in the command in general. 

The project graph is most valuable when you know which packages are brought through your project.